src/advanced-settings/utils.js

```diff
@@ -14,26 +14,29 @@ export default function validateAdvancedSettingsData(settingObj, setErrorFields)
     fieldsWithErrors.push({ key: settingName, message: 'Incorrectly formatted JSON' });
   };
 
-  const bracketsValidation = (value) => (/[{[\]}]/.test(value));
+  const bracketsValidation = (value) => /[{[\]}]/.test(value);
 
   Object.entries(settingObj).forEach(([settingName, settingValue]) => {
     const isArrayOrObject = (settingValue.startsWith('[') && settingValue.endsWith(']'))
         || (settingValue.startsWith('{') && settingValue.endsWith('}'));
-
+    const doubleAndSingleQuotes = /"(.*?)"|'(.*?)'/g;
+    const hasMatchingQuotes = doubleAndSingleQuotes.test(settingValue);
     if (typeof settingValue === 'string') {
       if (isArrayOrObject) {
         try {
           JSON.parse(settingValue);
         } catch (err) {
           pushDataToErrorArray(settingName);
         }
-      } else if (bracketsValidation(settingValue)) {
+      } else if (bracketsValidation(settingValue) || settingValue.includes('""') || settingValue.includes("''")
+      // eslint-disable-next-line no-mixed-operators
+      || settingValue.includes('"') && !hasMatchingQuotes || settingValue.includes("'") && !hasMatchingQuotes) {
```

Comment:
can we move the single/double quotes validation into a separate function?

Reply:
Thanks